hack/make-rules/test-e2e-node.sh

```diff
@@ -116,7 +116,7 @@ if [ ${remote} = true ] ; then
   IFS=',' read -ra IM <<< "${images}"
        images=""
        for i in "${IM[@]}"; do
-         if [[ $(gcloud compute instances list "${instance_prefix}-${i}" | grep ${i}) ]]; then
+         if gcloud compute instances list "${instance_prefix}-${i}" | grep "${i}"; then
```

Comment:
Why did it change and what did it fix:
```
In ./hack/make-rules/test-e2e-node.sh line 119:
         if [[ $(gcloud compute instances list "${instance_prefix}-${i}" | grep ${i}) ]]; then
               ^-- SC2143: Use grep -q instead of comparing output with [ -n .. ].
                                                                                ^--^ SC2086: Double quote to prevent globbing and word splitting.
Ref: https://github.com/koalaman/shellcheck/wiki/SC2143
```
Example
**Problematic code**:
```
if [ "$(find . | grep 'IMG[0-9]')" ]
then
  echo "Images found"
fi
```
**Correct code**:
```
if find . | grep -q 'IMG[0-9]'
then
  echo "Images found"
fi
```

Reply:
https://github.com/koalaman/shellcheck/wiki/SC2143#exceptions

most of our code uses pipefile, so piping to grep -q is not safe. using a fifo is safe however.